tests/time_tests/src/timetests/timetest_infer.cpp

```diff
@@ -17,59 +17,84 @@ using namespace InferenceEngine;
  * main(). The function should not throw any exceptions and responsible for
  * handling it by itself.
  */
-int runPipeline(const std::string &model, const std::string &device) {
-  auto pipeline = [](const std::string &model, const std::string &device) {
+int runPipeline(const std::string &model, const std::string &device, const std::string &performanceHint,
+                const bool isCacheEnabled, const std::string &vpuCompiler) {
+  auto pipeline = [](const std::string &model, const std::string &device, const std::string &performanceHint,
+                     const bool isCacheEnabled, const std::string &vpuCompiler) {
     Core ie;
     CNNNetwork cnnNetwork;
     ExecutableNetwork exeNetwork;
     InferRequest inferRequest;
     size_t batchSize = 0;
 
+    // enables performance hint for specified device
+    std::string performanceConfig;
+    if (performanceHint == "THROUGHPUT")
+      performanceConfig = CONFIG_VALUE(THROUGHPUT);
+    else if (performanceHint == "LATENCY")
+      performanceConfig = CONFIG_VALUE(LATENCY);
+    ie.SetConfig({{CONFIG_KEY(PERFORMANCE_HINT), performanceConfig}}, device);
```

Comment:
Is it possible to add check that 'PERFORMANCE_HINT' config is supported by device? Otherwise it will throw exception. Previously, I was able to use time_tests tool for GNA and MYRIAD which don't support such hints.
Due to same reason '-p' can be made optional

Reply:
added perf hint option check for specified device